src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -671,7 +671,31 @@ public GHAuthorization createToken(Collection<String> scope, String note, String
 
         return requester.method("POST").to("/authorizations", GHAuthorization.class).wrap(this);
     }
-
+    /**
+     * Creates a new authorization using an OTP.
+     * 
+     * Start by running createToken, if exception is thrown, prompt for OTP from user
+     * 
+     * Once OTP is received, call this token request
+     *
+     * The token created can be then used for {@link GitHub#connectUsingOAuth(String)} in the future.
+     *
+     * @see <a href="http://developer.github.com/v3/oauth/#create-a-new-authorization">Documentation</a>
+     */
+    public GHAuthorization createToken(Collection<String> scope, String note, String noteUrl, OTPReceiverLambda OTP) throws IOException{
```

Comment:
Add JavaDoc for the `@param`s, @return` and `@since`. 

And just a thought: 
```suggestion
    public GHAuthorization createToken(Collection<String> scope, String note, String noteUrl, Supplier<String> otpSupplier) throws IOException{
```

Reply:
yeah, i made this change, thanks for the heads up the generic existed!